build.gradle

```diff
@@ -63,9 +64,13 @@ task buildAwsLc {
             workingDir awslcSrcPath
             executable 'cmake'
             args "-B${cMakeBuildDir}"
-            args '-DBUILD_SHARED_LIBS=OFF'
+            args '-DBUILD_SHARED_LIBS=ON'
```

Comment:
Should building the shared library be contingent upon `isFIps`.  Is there a reason to prefer static over shared libraries in this context?

Reply:
Moving some conversation into this thread:

> I'm removing all static-linking logic as part of the FIPS work. This is for the following reasons:
> 1. We must support dynamic linking of libcrypto for FIPS
> 2. Supporting more than one linking mode increases complexity
> 3. The new CI (and potentially other systems) breaks statically linking an object into a shared object. The tests should be failing but are not. This is because ACCP (and some of our tests) are too robust. I'll fix this.

As follow ups from that:
1. I "fixed" the tests so that they properly fail when linking (static or dynamic) isn't don't quite right.
2. I think that I figured out why the new CI tests were not doing proper static linking but since it simplifies things to consolidate on a single linking strategy, that is not mot.

Static linking has two benefits and two drawbacks.
**Pros:**
* Smaller binary as we can remove unused symbols
* Easier loading story as everything ends up within a single shared object.

**Cons:**
* Cannot be used with AWS-LC FIPS
* More complicated (and fragile) build process.

Since we *must* support dynamic linking, it makes sense for us to reduce complexity by just doing that one mode instead of both.